package br.com.fiap.soat.techChallenge.exceptions;

import org.junit.jupiter.api.Test;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ClienteNaoEncontradoExceptionTest {

    @Test
    void testCriacaoAPartirDoCpf() {
        // Dado
        String cpf = "123.456.789-01";

        // Quando
        ClienteNaoEncontradoException exception = ClienteNaoEncontradoException.aPartirDoCpf(cpf);

        // Ent達o
        assertEquals("O cliente de CPF 123.456.789-01 n達o foi encontrado.", exception.getMessage());
    }

    @Test
    void testCriacaoAPartirDoId() {
        // Dado
        UUID id = UUID.randomUUID();

        // Quando
        ClienteNaoEncontradoException exception = ClienteNaoEncontradoException.aPartirDoId(id);

        // Ent達o
        assertEquals(String.format("O cliente de ID %s n達o foi encontrado.", id), exception.getMessage());
    }
}
